keps/sig-api-machinery/4420-retry-generate-name/README.md

```diff
@@ -0,0 +1,525 @@
+
+No
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+This can happen.
+
+When a retry does occur, if the resource being created with generateName has
+validating webhooks registered, the validating webhooks will be called for each
+retry. For slow webhooks this could result in a significant increase to request
+latency for the create requests using generateName.
+
+We expect retries to be very rare, even for large numbers of resources
+(100k-1M), but if a cluster were somehow able to generate a very large number of
+resources for a generateName, say 2M or higher, then retries could become more
+common. 
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No, at least not in all but the most pathologcial of situations.
+
+If a cluster was receiving a high load of create requests for generateName
+resources with a large number (millions) of existing resources already generated
+with the same generateName prefix, this could cause up to a 7x increase in CPU.
+But we do not expect this to actually happen, primarily because etcd cannot
+support these volumes of resources.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+###### What are other known failure modes?
+
+The only know failure modes are pathological. E.g. A runaway system creating
+unbounded resources using generateName could eventually trigger amplified load
+on webhooks with this feature.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+Look at the number of resources (using kube-state-metrics or similar). If there
+exists a very large number of resources (millions) of a particular resource kind
+and they all share the same generateName prefix, this might be triggering
+high retry load.
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+This code is in a critical system codepath, if anything is wrong with it, it could cause
+severe problems, up to and including data corruption and loss.
+
+## Alternatives
+
+There are many alternatives that we considered including:
+
+- Increasing the number of characters appended to generateName when generating
```

Comment:
I assume number of chars is the length, but what about the number of possible chars? is that an option?

Reply:
It's an alternative. But it is a breaking change in two ways:

- there is a 63 char name length limit. With the 5 char random suffix length, a 58 char length limit is set on generateName. We'd have to change that (or change the max length limit name!) to make room for a change in random suffix length.
- there are major downstream projects that use "{5}" regexes to match the random suffix.

Reply:
I've clarified in the KEP